src/Type/Doctrine/QueryBuilder/CreateQueryBuilderDynamicReturnTypeExtension.php

```diff
@@ -48,7 +48,8 @@ public function getTypeFromMethodCall(
 		}
 
 		return new $class(
-			$this->queryBuilderClass ?? 'Doctrine\ORM\QueryBuilder'
+			$this->queryBuilderClass ?? 'Doctrine\ORM\QueryBuilder',
+			$scope
```

Comment:
To pass Scope in a type is weird. Perhaps we should do `argumentsProcessor->processArgs` sooner and carry the final args in the Type? Or something different like that.

Reply:
That was my original thought, but this felt easier. Anyway, do you see any real issue with this approach?

Reply:
Btw to fix the failing tests, I probably need to store Scope per the method call inside.

Reply:
I feel like this is still going to be wrong in some cases, as the failing tests prove. One Scope per QueryBuilder is not enough as the scope changes when the QB gets built. Better process the args sooner.